azure-proto-core/ArmClientOptions.cs

```diff
@@ -1,14 +1,34 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.ComponentModel;
-
+using Azure.Core;
+using Azure.Core.Pipeline;
 namespace azure_proto_core
 {
-    public class ArmClientOptions
+    public class ArmClientOptions : ClientOptions
     {
         private Dictionary<Type, object> _overrides = new Dictionary<Type, object>();
         private static readonly object _overridesLock = new object();
 
+        //TODO policy lists are internal hence we don't have acces to them by inheriting ClientOptions in this Asembly, this is a wrapper for now to convert to the concrete
+        //policy options. 
+        public new void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position){
```

Comment:
Why is this necessary?  AddPolicy(...) is public so we shouldn't need access to the direct lists themselves do we?

Reply:
Because we need to access list to copy over the polices for the concrete types

https://github.com/m-nash/azure-proto-sdk/blob/de6bee33ea5f092501d7283359db69d214ce5a51/azure-proto-core/SubscriptionContainerOperations.cs#L58